src/pdm/project/core.py

```diff
@@ -148,6 +149,10 @@ def project_config(self) -> Config:
     def name(self) -> str:
         return self.pyproject.metadata.get("name")
 
+    @cached_property
```

Comment:
`self.pyproject` is subject to change, don't cache this. Instead, you can add cache to the `normalize_name` function

Reply:
Done.